core/primitives-core/src/types.rs

```diff
@@ -18,6 +18,9 @@ pub type Nonce = u64;
 pub type BlockHeight = u64;
 /// Height of the epoch.
 pub type EpochHeight = u64;
+/// Block hash.
+#[cfg(feature = "protocol_feature_vm_hash")]
+pub type BlockHash = [u8; 32];
```

Comment:
Why is it a new type instead of a `CryptoHash`?

Reply:
Oh good catch. I did use it before finding `CryptoHash`. Forgot to remove it.

Reply:
`type` in rust introduce alias, `pub type BlockHash = CryptoHash` plus `type CryptoHash = [u8;32]` will have equivalent semantic result as `type BlockHash=[u8;32]` but more robust if some day we change CryptoHash's definition